src/components/Maps/SearchBar.tsx

```diff
@@ -0,0 +1,130 @@
+import React, { useState, useRef } from 'react';
+import Popover from '@material-ui/core/Popover';
+import List from '@material-ui/core/List';
+import ListItem from '@material-ui/core/ListItem';
+import Typography from '@material-ui/core/Typography';
+import IconButton from '@material-ui/core/IconButton';
+import InputAdornment from '@material-ui/core/InputAdornment';
+import SearchIcon from '@material-ui/icons/Search';
+import PlacesAutocomplete, {
+  geocodeByAddress,
+  getLatLng,
+} from 'react-places-autocomplete';
+import { TextValidator } from 'react-material-ui-form-validator';
+import { makeStyles } from '@material-ui/core/styles';
+import { Colours } from '../../styles/Constants';
+
+const useStyles = makeStyles({
+  sidebarOptions: {
+    width: '334px',
+    '&:hover': {
+      backgroundColor: Colours.BackgroundGray,
+    },
+  },
+  sidebarTextField: {
+    width: '334px',
+    marginLeft: '16px',
+  },
+  sidebarList: {
+    width: '324px',
+  },
+});
+
+const SearchBar = ({
+  onComplete,
+}: {
+  onComplete: ({ latitude, longitude, address }) => void;
+}) => {
+  const [address, setAddress] = useState('');
+  const [menuOpen, setMenuOpen] = useState(false);
+  const styles = useStyles();
+  const inputEl = useRef(null);
+
+  const searchOptions = {
+    componentRestrictions: {
+      country: ['CA'],
+    },
+  };
+
+  const handleChange = (address) => {
+    setAddress(address);
+  };
+
+  const handleSelect = (selectedAddress) => {
+    geocodeByAddress(selectedAddress)
+      .then((results) => getLatLng(results[0]))
+      .then((latLng) => {
+        onComplete({
+          latitude: latLng.lat,
+          longitude: latLng.lng,
+          address: selectedAddress,
+        });
+      });
+    setMenuOpen(false);
+    setAddress(selectedAddress);
+  };
+
+  return (
+    <PlacesAutocomplete
+      value={address}
+      onChange={handleChange}
+      onSelect={handleSelect}
+      searchOptions={searchOptions}
+    >
+      {({ getInputProps, suggestions, getSuggestionItemProps, loading }) => (
```

Comment:
i might've missed it, but where do we define `suggestions` and `getSuggestionItemProps`? Or is this provided by the library we're using?

Reply:
This is from the `react-places-autocomplete` library